lib/rbeapi/api/alias.rb

```diff
@@ -0,0 +1,159 @@
+#
+## Copyright (c) 2016, Arista Networks, Inc.
+## All rights reserved.
+##
+## Redistribution and use in source and binary forms, with or without
+## modification, are permitted provided that the following conditions are
+## met:
+##
+##   Redistributions of source code must retain the above copyright notice,
+##   this list of conditions and the following disclaimer.
+##
+##   Redistributions in binary form must reproduce the above copyright
+##   notice, this list of conditions and the following disclaimer in the
+##   documentation and/or other materials provided with the distribution.
+##
+##   Neither the name of Arista Networks nor the names of its
+##   contributors may be used to endorse or promote products derived from
+##   this software without specific prior written permission.
+##
+## THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+## "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+## LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+## A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL ARISTA NETWORKS
+## BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+## CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+## SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+## BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+## WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+## OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+## IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+##
+require 'rbeapi/api'
+
+##
+# Rbeapi toplevel namespace.
+module Rbeapi
+  ##
+  # Api is module namespace for working with the EOS command API.
+  module Api
+    ##
+    # The Alias class manages aliass entries on an EOS node.
+    class Alias < Entity
+
+      ##
+      # get returns the current alias configuration hash extracted from the
+      # nodes running configuration.
+      #
+      # @example
+      #   {
+      #     alias: array<strings>
+      #   }
+      #
+      # @return [Hash<Symbol, Object>] Returns the alias resource as a hash
+      #   object from the nodes current configuration.
+      def get(name)
+        aliases = config.scan(/^alias #{name}((?:(?= )(?:.+?)(?=\n)|(?:\n(?:.+?))(?=\n\!)))/m)
```

Comment:
this regex is still not perfect, I have an extra space in the beginning of the first capture group